docs/modules/ROOT/pages/high_level/security_group_host_design.adoc

```diff
@@ -0,0 +1,466 @@
+
+[source,shell]
+------------------------------------------------------------
+   +-------------+                     +---------------+
+   |             |                     |               |
+   |    VM 1     |                     |     VM 2      |
+   |             |                     |               |
+   +---+------+--+                     +---+-------+---+
+       | eth  |                            |  eth  |
+       +--+---+                            +---+---+
+          |                                    |
+       +--+---+                            +---+---+
+       | tap  |                            |  tap  |
+ +-----+------+-----+                 +----+-------+-----+
+ |                  |                 |                  |
+ |   Linux Bridge 1 |                 |   Linux Bridge 2 |
+ |                  |                 |                  |
+ +-----+------+-----+                 +-----+------+-----+
+       | veth |                             | veth |
+       +--+---+                             +--+---+
+          |                                    |
+       +--+---+                             +--+---+
+       | veth |                             | veth |
++------+------+-----------------------------+------+---------+
+|                                                            |
+|                     OVS  Bridge  (br-int)                  |
+|                                                            |
++------------------------------------------------------------+
+------------------------------------------------------------
+
+
+=== After OpenStack Mitaka Release
+
+The OpenStack community recognized the challenge of IP Tables implementation for security group rules.
+With OVS implementation of connection tracking <<ovs-conntrack>>, many open source solutions including ovs_dpdk_firewall<<intel>> and Dragonflow<<dragonflow-security-group>> start leveraging it for OpenStack.
+It is natural for OpenStack Neutron to adopt it.
+OpenStack Mitaka introduced official OVS Firewall driver, which addressed many of the issues in the prior releases.
+But Alcor can do even better than that.
+
+
+== Design approach
+
+The Security Group design uses OVS and OpenFlow, building on the latest Neutron OVS firewall rule approach.
+It offers the following advantages compared to Neutron:
+
+. On-Host direct connection from VM port to OVS: it doesn't need extra Linux bridge, Linux devices, Linux kernel or IP Tables;
+. Direct and simplified communication between Alcor controller to Alcor control agent;
+. Delta update: Update of security group rule only require the changed rule to be push down to compute hosts;
+. Reduced OVS SG rules: Only one set of security group rules is configured in OVS to be shared with all ports using it
+. Enable scale and performance from the ground up, addressing the biggest pain point on Neutron
```

Comment:
Is the pain point of neutron remote_group or is it that multiple vm on the same host share a security group rule?

Reply:
I believe one of the pain point is having too many openflow rules installed on a compute node. Both neutron remote_group and mutiple VMs (on same host) assoicated to a lot of security group rules would contribute to the pain point.